dvc/tree/webhdfs.py

```diff
@@ -0,0 +1,143 @@
+import logging
+import os
+import threading
+from contextlib import contextmanager
+from urllib.parse import urlparse
+
+from funcy import cached_property, wrap_prop
+
+from dvc.hash_info import HashInfo
+from dvc.path_info import CloudURLInfo
+from dvc.progress import Tqdm
+from dvc.scheme import Schemes
+
+from .base import BaseTree
+
+logger = logging.getLogger(__name__)
+
+
+def update_pbar(pbar):
+    """Update pbar to accept the two arguments passed by hdfs"""
+
+    def update(_, bytes_transfered):  # No use for first argument (path)
+        if bytes_transfered == -1:
+            return
+        pbar.update(bytes_transfered)
+
+    return update
+
+
+class WebHDFSTree(BaseTree):
+    scheme = Schemes.WEBHDFS
+    PATH_CLS = CloudURLInfo
+    REQUIRES = {"hdfs": "hdfs"}
+    PARAM_CHECKSUM = "checksum"
+
+    def __init__(self, repo, config):
+        super().__init__(repo, config)
+
+        self.path_info = None
+        url = config.get("url")
+        if not url:
+            return
+
+        parsed = urlparse(url)
+        user = parsed.username or config.get("user")
+
+        self.path_info = self.PATH_CLS.from_parts(
+            scheme="webhdfs",
+            host=parsed.hostname,
+            user=user,
+            port=parsed.port,
+            path=parsed.path,
+        )
+
+        self.hdfscli_config = config.get("hdfscli_config")
+        self.token = config.get("webhdfs_token")
+        self.alias = config.get("webhdfs_alias")
+
+    @wrap_prop(threading.Lock())
+    @cached_property
+    def hdfs_client(self):
+        import hdfs
+
+        logger.debug("URL: %s", self.path_info)
+        logger.debug("HDFSConfig: %s", self.hdfscli_config)
+
+        try:
+            client = hdfs.config.Config(self.hdfscli_config).get_client(
+                self.alias
+            )
```

Comment:
Really minor: Let's just return here and below, since we are not really doing anything with the `client`.
```suggestion
            return hdfs.config.Config(self.hdfscli_config).get_client(
                self.alias
            )
```